Description: Fix build system for Emscripten cross-compilation.
Forwarded: no
Author: Sylvain Beucler <beuc@beuc.net>
Last-Update: 2020-05-08

Index: Python-3.8.3/config.sub
===================================================================
--- Python-3.8.3.orig/config.sub
+++ Python-3.8.3/config.sub
@@ -1394,7 +1394,7 @@ case $os in
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
-	      | -midnightbsd*)
+	      | -midnightbsd* | -emscripten*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
Index: Python-3.8.3/configure
===================================================================
--- Python-3.8.3.orig/configure
+++ Python-3.8.3/configure
@@ -3278,6 +3278,9 @@ then
 	*-*-vxworks*)
 	    ac_sys_system=VxWorks
 	    ;;
+	*-*-emscripten)
+		ac_sys_system=Emscripten
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -3328,6 +3331,9 @@ if test "$cross_compiling" = yes; then
 	*-*-vxworks*)
 		_host_cpu=$host_cpu
 		;;
+	*-*-emscripten)
+		ac_sys_system=Emscripten
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
Index: Python-3.8.3/config-emscripten.site
===================================================================
--- /dev/null
+++ Python-3.8.3/config-emscripten.site
@@ -0,0 +1,5 @@
+# Required by ./configure
+ac_cv_file__dev_ptmx=no
+ac_cv_file__dev_ptc=no
+# Support dynamic linking
+ac_cv_func_dlopen=yes
